includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -135,6 +135,11 @@ public function register_scripts() {
 			WC_Payments::get_file_version( 'dist/upe_checkout.js' ),
 			true
 		);
+
+		if ( ! WC()->cart->needs_payment() && is_checkout() && ! has_block( 'woocommerce/checkout' ) ) {
+			WC_Payments::get_gateway()->tokenization_script();
+			WC_Payments::get_wc_payments_checkout()->enqueue_payment_scripts();
+		}
```

Comment:
IMO, this whole block can be moved to a method in `WC_Payment_Gateway_WCPay` and then you can use that method for all of these places. Without that, we do not need to write the same code in three places. We can use this because of the inheritance:  

- UPE_Payment_Gateway extends WC_Payment_Gateway_WCPay
- UPE_Split_Payment_Gateway extends UPE_Payment_Gateway

Another possibility is to just hook this extracted method into the `wp_enqueue_scripts` filter. With that, you just need to write this code one time in the top-level class `WC_Payment_Gateway_WCPay`.

Something like this: https://gist.github.com/htdat/679c224b1c5a761be6082e44695b76ee